### DISCUSSION TITLE
What are the  PlanInstruction profiles available

### DISCUSSION THREAD

**Discussion by mohamedsayed18** [tesseract-robotics/tesseract_planning#105] [Q&A] [answered]
for the third parameter in the constructor `FREESPACE` what are the other available and what is the difference between them?
```
PlanInstruction plan_f0(wp1, PlanInstructionType::FREESPACE, "FREESPACE");
```
Does this mean every instruction can have its own planning profile?

**Comment by Levi-Armstrong** [comment, marked as answer]
It is a key that corresponds to a profile to be used for configuring planners. For example, you can create a profile for Descartes, TrajOpt, and OMPL which get stored with the Key "FREESPACE" and then when you leverage one of the process planners it will use this key to look up the profile for configuring the planner.

**Reply by mohamedsayed18** [reply]
Does this mean every instruction can have its own planning profile?

**Reply by Levi-Armstrong** [reply]
That is correct.
